common/internal_model/src/main/scala/weco/catalogue/internal_model/index/WorksAnalysis.scala

```diff
@@ -8,12 +8,11 @@ object WorksAnalysis {
   val slashesCharFilter =
     MappingCharFilter("slashes_char_filter", mappings = Map("/" -> " __"))
 
-  // This analyzer "keeps" the hyphen, by removing it and treating hyphenated
-  // tokens as a single token.
-  val hyphensCharFilter =
-    PatternReplaceCharFilter(
-      "hyphens_char_filter",
-      pattern = "-",
+  // This char filter removes all punctuation
+  val punctuationTokenFilter =
+    PatternReplaceTokenFilter(
+      "punctuation",
+      pattern = "[^\\w\\s]",
```

Comment:
According to the [Java documentation](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html), `\w` means _A word character: `[a-zA-Z_0-9]`_ – if taken literally, that would filter out any accented characters, which might cause us issues.

Do we need to worry about accented characters being filtered out here, or is this a non-issue?

Reply:
great catch! 

does `[^0-9\\p{L}\\s]` look right to you?

Reply:
## Using the [^\\w\\s] pattern

```json
GET /_analyze
{
  "tokenizer": "standard",
  "filter": [
    {
      "type": "pattern_replace",
      "pattern": "[^\\w\\s]",
      "replacement": ""
    }
  ],
  "text": "C'est l'été! Comment ça va? J'ai 200€here's some arabic text عرف"
}
```

```json
{
  "tokens": [
    {
      "token": "Cest",
      "start_offset": 0,
      "end_offset": 5,
      "type": "<ALPHANUM>",
      "position": 0
    },
    {
      "token": "lt",
      "start_offset": 6,
      "end_offset": 11,
      "type": "<ALPHANUM>",
      "position": 1
    },
    {
      "token": "Comment",
      "start_offset": 13,
      "end_offset": 20,
      "type": "<ALPHANUM>",
      "position": 2
    },
    {
      "token": "a",
      "start_offset": 21,
      "end_offset": 23,
      "type": "<ALPHANUM>",
      "position": 3
    },
    {
      "token": "va",
      "start_offset": 24,
      "end_offset": 26,
      "type": "<ALPHANUM>",
      "position": 4
    },
    {
      "token": "Jai",
      "start_offset": 28,
      "end_offset": 32,
      "type": "<ALPHANUM>",
      "position": 5
    },
    {
      "token": "200",
      "start_offset": 33,
      "end_offset": 36,
      "type": "<NUM>",
      "position": 6
    },
    {
      "token": "heres",
      "start_offset": 37,
      "end_offset": 43,
      "type": "<ALPHANUM>",
      "position": 7
    },
    {
      "token": "some",
      "start_offset": 44,
      "end_offset": 48,
      "type": "<ALPHANUM>",
      "position": 8
    },
    {
      "token": "arabic",
      "start_offset": 49,
      "end_offset": 55,
      "type": "<ALPHANUM>",
      "position": 9
    },
    {
      "token": "text",
      "start_offset": 56,
      "end_offset": 60,
      "type": "<ALPHANUM>",
      "position": 10
    },
    {
      "token": "",
      "start_offset": 61,
      "end_offset": 64,
      "type": "<ALPHANUM>",
      "position": 11
    }
  ]
}
```

</details>


## Using the [^0-9\\p{L}\\s] pattern

```json
GET /_analyze
{
  "tokenizer": "standard",
  "filter": [
    {
      "type": "pattern_replace",
      "pattern": "[^0-9\\p{L}\\s]",
      "replacement": ""
    }
  ],
  "text": "C'est l'été! Comment ça va? J'ai 200€here's some arabic text عرف"
}
```

```json
{
  "tokens": [
    {
      "token": "Cest",
      "start_offset": 0,
      "end_offset": 5,
      "type": "<ALPHANUM>",
      "position": 0
    },
    {
      "token": "lété",
      "start_offset": 6,
      "end_offset": 11,
      "type": "<ALPHANUM>",
      "position": 1
    },
    {
      "token": "Comment",
      "start_offset": 13,
      "end_offset": 20,
      "type": "<ALPHANUM>",
      "position": 2
    },
    {
      "token": "ça",
      "start_offset": 21,
      "end_offset": 23,
      "type": "<ALPHANUM>",
      "position": 3
    },
    {
      "token": "va",
      "start_offset": 24,
      "end_offset": 26,
      "type": "<ALPHANUM>",
      "position": 4
    },
    {
      "token": "Jai",
      "start_offset": 28,
      "end_offset": 32,
      "type": "<ALPHANUM>",
      "position": 5
    },
    {
      "token": "200",
      "start_offset": 33,
      "end_offset": 36,
      "type": "<NUM>",
      "position": 6
    },
    {
      "token": "heres",
      "start_offset": 37,
      "end_offset": 43,
      "type": "<ALPHANUM>",
      "position": 7
    },
    {
      "token": "some",
      "start_offset": 44,
      "end_offset": 48,
      "type": "<ALPHANUM>",
      "position": 8
    },
    {
      "token": "arabic",
      "start_offset": 49,
      "end_offset": 55,
      "type": "<ALPHANUM>",
      "position": 9
    },
    {
      "token": "text",
      "start_offset": 56,
      "end_offset": 60,
      "type": "<ALPHANUM>",
      "position": 10
    },
    {
      "token": "عرف",
      "start_offset": 61,
      "end_offset": 64,
      "type": "<ALPHANUM>",
      "position": 11
    }
  ]
}
```

</details>